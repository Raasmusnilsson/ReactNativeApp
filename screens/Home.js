import { View, Text, Button, Image, TextInput } from 'react-native'
import React, { useState } from 'react'
import logo from '../assets/meme-logo.png'
import ScreenTitle from '../components/ScreenTitle'

function Home({ navigation }) {
	const [name, setName] = useState('')
	const [typedName, setTypedName] = useState(false)

	return (
		<View style={{ flex: 1, alignItems: 'center', justifyContent: 'top' }}>
			<ScreenTitle message="Classic Meme Images" />

			<Image
				resizeMode="contain"
				source={logo}
				style={{ width: '75%', height: '75%' }}
			/>

			<TextInput
				style={{ borderWidth: 1, borderRadius: 1 }}
				placeholder="What's your name?"
				onChange={(event) => setName(event.target.value)}
			/>

			{name.length > 2 && typedName === false && (
				<Button title="Get a greeting" onPress={() => setTypedName(true)} />
			)}

			{typedName && (
				<Text>
					Welcome, {name}! Continue to the{' '}
					<Button title="About" onPress={() => navigation.navigate('About')} />{' '}
					page!
				</Text>
			)}
		</View>
	)
}
export default Home
