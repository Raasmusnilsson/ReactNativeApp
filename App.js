import * as React from 'react'
import { NavigationContainer } from '@react-navigation/native'
import { createDrawerNavigator } from '@react-navigation/drawer'
import Home from './screens/Home'
import Memes from './screens/Memes'
import About from './screens/About'
import APIContextProvider from './components/APIContext'

const Drawer = createDrawerNavigator()

function App() {
	return (
		<APIContextProvider>
			<NavigationContainer>
				<Drawer.Navigator initialRouteName="Home">
					<Drawer.Screen name="Home" component={Home} />
					<Drawer.Screen name="Memes" component={Memes} />
					<Drawer.Screen name="About" component={About} />
				</Drawer.Navigator>
			</NavigationContainer>
		</APIContextProvider>
	)
}

export default App
