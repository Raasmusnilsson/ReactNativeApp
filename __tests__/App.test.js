import renderer from 'react-test-renderer'
import * as React from 'react'
import 'react-native'
import App from '../App'
import { render } from '@testing-library/react-native'
import Home from '../screens/Home'

it('renders correctly', () => {
	renderer.create(<App />)
})

it('have right placeholder text', () => {
	const { getByPlaceholderText } = render(<Home />)
	const element = getByPlaceholderText("What's your name?")
})

it('matches snapshot', () => {
	const tree = renderer.create(<App />).toJSON()
	expect(tree).toMatchSnapshot()
})

it('get element by id', () => {
	const { queryAllByTestId } = render(<Home />)
	const element = queryAllByTestId('greeting')
})
