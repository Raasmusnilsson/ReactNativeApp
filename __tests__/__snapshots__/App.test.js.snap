// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "top",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "top",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 25,
        }
      }
    >
      Classic Meme Images
    </Text>
  </View>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../assets/meme-logo.png",
      }
    }
    style={
      Object {
        "height": "75%",
        "width": "75%",
      }
    }
  />
  <TextInput
    onChange={[Function]}
    placeholder="What's your name?"
    style={
      Object {
        "borderRadius": 1,
        "borderWidth": 1,
      }
    }
  />
</View>
`;
